dvc/dependency/repo.py

```diff
@@ -70,8 +70,13 @@ def download(self, to, resume=False):
             out = repo.find_out_by_relpath(self.def_path)
             with repo.state:
                 repo.cloud.pull(out.get_used_cache())
-            to.info = copy.copy(out.info)
-            to.checkout()
+
+        return out
+
+    def download(self, to):
+        out = self.fetch()
+        to.info = copy.copy(out.info)
```

Comment:
Mutating an argument can bite as some day. Is there a better way?

Reply:
Sure there is, might want to create a ticket for it. But let's be clear that it is outside of this PR's scope.